File: programmers/electric.java
import java.util.*;

class Solution {
    static int arr[][];
    // ArrayList<Integer>[] arr = new ArrayList[100];
    
    public int solution(int n, int[][] wires) {
        int answer = n;
        arr = new int[n+1][n+1];
        
        for(int i = 0; i < wires.length; i++){
            arr[wires[i][0]][wires[i][1]] = 1;
            arr[wires[i][1]][wires[i][0]] = 1;
        }
        
        for(int i = 0; i < wires.length; i++){
            arr[wires[i][0]][wires[i][1]] = 0;
            arr[wires[i][1]][wires[i][0]] = 0;       
            
            answer = Math.min(answer, bfs(n, wires[i][0]));
            
            arr[wires[i][0]][wires[i][1]] = 1;
            arr[wires[i][1]][wires[i][0]] = 1;
        }
        
        return answer;
    }
    
    public int bfs(int n, int start){
        int[] visit = new int[n+1];
        int cnt = 1;
        
        Queue<Integer> q = new LinkedList<>();
        q.offer(start);
        
        while(!q.isEmpty()){
            int point = q.poll();
            visit[point] = 1;
            
            for(int i = 1; i <= n; i++){
                if(arr[point][i] == 1 && visit[i] == 0){
                    q.offer(i);
                    cnt++;
                }
            }
        }
        return (int)Math.abs(n-2 * cnt);
    }
}

File: programmers/electric.cpp
#include <bits/stdc++.h>

using namespace std;

int cnt, groupCnt;
bool check[100+1];
bool arr[100+1][100+1];
int nums[2];

void dfs(int start, int n){
    check[start] = true;
    nums[groupCnt]++;
    for(int i = 1; i <= n; i++)
        if(arr[start][i] && !check[i])
            dfs(i, n);
}

int solution(int n, vector<vector<int>> wires) {
    int ans = 987654321;
    for(int i = 0; i < wires.size(); i++){
        arr[wires[i][0]][wires[i][1]] = true;
        arr[wires[i][1]][wires[i][0]] = true;
    }
    
    for(int i = 0; i < wires.size(); i++){
        arr[wires[i][0]][wires[i][1]] = false;
        arr[wires[i][1]][wires[i][0]] = false;    
        for(int j = 1; j <= n; j++){
            if(check[j]) continue;
            dfs(j, n);
            groupCnt++;
        }
        arr[wires[i][0]][wires[i][1]] = true;
        arr[wires[i][1]][wires[i][0]] = true;    
        
        if(groupCnt == 2)
            ans = min(ans, abs(nums[1] - nums[0]));
        
        groupCnt = 0;
        nums[0] = nums[1] = 0;
        memset(check, false, sizeof(check));
    }    
    return ans;
}
